sr_logging_common/logging/check_log_size.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import shutil
+import rospy
+import logging_utils as utils
+
+
+if __name__ == '__main__':
+    rospy.init_node('check_log_size')
+
+    additional_required_disk_space = rospy.get_param('~additional_required_disk_space', 10) * utils.GIGABYTE
+    allocated_log_space = rospy.get_param('~allocated_log_space', 10) * utils.GIGABYTE
+    log_growth_headroom = rospy.get_param('~log_growth_headroom', 5) * utils.GIGABYTE
+
+    total_disk_space, _, free_disk_space = shutil.disk_usage(utils.LOG_PATH)
+    logs_size = utils.get_directory_size(utils.LOG_PATH)
+
+    if free_disk_space < log_growth_headroom + max(logs_size, allocated_log_space) + additional_required_disk_space:
+        required_disk_space = log_growth_headroom + max(logs_size, allocated_log_space) + additional_required_disk_space
+        rospy.logfatal("Not enough free disk space to safely store roslogs:\n" +
+                       f"\tFree Disk Space: {free_disk_space / utils.GIGABYTE:.3f} of"
+                       f" {total_disk_space / utils.GIGABYTE:.3f} GB free\n" +
+                       f"\tRequired Free Disk Space: {required_disk_space / utils.GIGABYTE:.3f} GB\n\t\tMinimum" +
```

Comment:
It's a bit confusing that there's a "Required Free Disk Space" and a "Minimum Required Free Disk Space" and they're not the same thing.
One would expect that the requirement would be for a minimum space. What does this mean?

Reply:
maybe non_log_headroom?

Reply:
I still think this shouldn't be printed as "Minimum Required Free Disk Space" though.

Reply:
https://github.com/shadow-robot/common_resources/pull/265/commits/b0f7deb1b0e4752bc730d5720b7b737e726eadd8